File: codes/easy/2073 Time Needed to Buy Tickets/discussion.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int count = 0;

        for(int i = 0; i < tickets.length; i++) {
            if(i <= k) {
                count += Math.min(tickets[k], tickets[i]);
            } else {
                count += Math.min(tickets[k] - 1, tickets[i]);
            }
        }

        return count;
    }
}

File: codes/easy/2073 Time Needed to Buy Tickets/discussion.js
/**
 * @param {number[]} tickets
 * @param {number} k
 * @return {number}
 */
var timeRequiredToBuy = function(tickets, k) {
    let count = 0;

    for(let i = 0; i < tickets.length; i++) {
        count += Math.min(tickets[k] - (k < i), tickets[i]);
    }

    return count;
};